protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -116,6 +116,7 @@ impl MovementClient {
 		})
 	}
 
+	#[cfg(feature = "bridge-tests")]
```

Comment:
To avoid putting the test code in the prod code you can use 2 impl like : 
for the prod code
```
impl MovementClient {
  pub async fn new(_config: Config) -> Result<Self, anyhow::Error> {
    ....
}
```
And later in the code for the test, and you feature flag the whole impl.
```
#[cfg(feature = "bridge-tests")]
impl MovementClient {
   pub async fn new_for_test(....
}
```
You can do the same for config also.

Reply:
Thanks @musitdev, can the impls with the feature flag go in the same file (protocol-units/bridge/chains/movement/src/lib.rs) or is it better to refactor into a file specifically for tests?

Reply:
I don't know, there's no best way. My opinion is that we can leave them in the same file.

Reply:
Same file is fine yes. But move the impl block for testing down, beneath the more important methods, as its a long method.